var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -61,31 +63,38 @@ class Hdf5(AutotoolsPackage):
     variant('java', default=False, description='Enable Java support')
     variant('threadsafe', default=False,
             description='Enable thread-safe capabilities')
-
+    variant('tools', default=False, description='Enable build tools')
     variant('mpi', default=True, description='Enable MPI support')
     variant('szip', default=False, description='Enable szip support')
+    variant('zlib', default=True, description='Enable zlib support')
     variant('pic', default=True,
             description='Produce position-independent code (for shared libs)')
-    # Build HDF5 with API compaitibility.
-    variant('api', default='none', description='choose api compatibility', values=('none', 'v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
+    # Build HDF5 with API compatibility.
+    variant('api', default='none', description='Choose api compatibility for earlier version', values=('none', 'v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
 
     conflicts('api=v114', when='@1.6:1.12.99', msg='v114 is not compatible with this release')
     conflicts('api=v112', when='@1.6:1.10.99', msg='v112 is not compatible with this release')
     conflicts('api=v110', when='@1.6:1.8.99', msg='v110 is not compatible with this release')
     conflicts('api=v18', when='@1.6:1.6.99', msg='v18 is not compatible with this release')
 
-    depends_on('autoconf', type='build', when='@develop')
-    depends_on('automake', type='build', when='@develop')
-    depends_on('libtool',  type='build', when='@develop')
-    depends_on('m4',       type='build', when='@develop')
+    depends_on('cmake@3.12:')
+    depends_on('readline')
+    depends_on('gdbm')
+    depends_on('perl')
```

Comment:
I would undo these last couple of commits. CMake should be a build dependency. If readline/gdbm/perl were only added to fix `spack graph`, then that's a bug in `spack graph`, not a bug in the package definition.